src/test/java/io/airbrake/javabrake/RouteTest.java

```diff
@@ -0,0 +1,162 @@
+    RouteMetric metric = new RouteMetric("GET", "/test");
+
+    Metrics.FLUSH_PERIOD = 5;
+    metric.endTime = new Date();
+
+    String date = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX").format(metric.startTime);
+    RouteStats stat = new RouteStats(
+        metric.method,
+        metric.route,
+        metric.statusCode,
+        date);
+
+    long ms = metric.endTime.getTime() - metric.startTime.getTime();
+    stat.add(ms);
+    return stat;
+  }
+
+  @Test
+  @Order(1)
+  public void testPerformanceStatsFalse() {
+
+    config.performanceStats = false;
+    RouteMetric metric = new RouteMetric("GET", "/test");
+    metric.statusCode = 200;
+    metric.contentType = "application/json";
+    try {
+      notifier.routes.notify(metric);
+    } finally {
+      assertEquals(Routes.status, "performanceStats is disabled");
+    }
+  }
+
+  @Test
+  @Order(2)
+  public void testRoutesDefaultEnv() {
+
+    config.performanceStats = true;
+    config.environment = null;
+    RouteMetric metric = new RouteMetric("GET", "/test");
+    metric.statusCode = 200;
+    metric.contentType = "application/json";
+    try {
+      notifier.routes.notify(metric);
+    } finally {
+      assertEquals(config.environment, "production");
+    }
+  }
+
+  @Test
+  @Order(3)
+  public void testRouteNotifyException() {
+
+    config.performanceStats = true;
+    config.projectId = 1;
+    notifier.setAPMHost("http://localhost:8080");
+
+    stubFor(post(urlEqualTo("/api/v5/projects/1/routes-stats")).withHeader("Authorization", containing("Bearer "))
+        .willReturn(aResponse().withBody("{}")
+            .withStatus(200)));
+
+    try {
+      notifier.routes.notify(null);
+
+    } finally {
+      assertEquals(Routes.status, "java.lang.NullPointerException");
+    }
+  }
+
+  @Test
+  @Order(4)
+  public void testRouteNotifySuccess() {
+
+    config.performanceStats = true;
+    config.projectId = 1;
+    notifier.setAPMHost("http://localhost:8080");
+
+    stubFor(post(urlEqualTo("/api/v5/projects/1/routes-stats")).withHeader("Authorization", containing("Bearer "))
+        .willReturn(aResponse().withBody("{'message':'Success'}")
+            .withStatus(200)));
+
+    List<Object> routeList = new ArrayList<>();
+    routeList.add(getRouteStats());
+    Routes routes = new Routes(Notifier.config.environment, routeList);
+
+    OkSender okSender = new OkSender(config);
+
+    Call call = OkSender.okhttp.newCall(okSender.buildAPMRequest(OkSender.gson.toJson(routes), Constant.apmRoute));
+    try (Response resp = call.execute()) {
+      JSONObject res = null;
+
+      try {
+        Buffer buffer = new Buffer();
+        try {
+          resp.request().body().writeTo(buffer);
+        } catch (IOException e) {
+          // TODO Auto-generated catch block
+          e.printStackTrace();
+        }
+        String json = buffer.readUtf8();
+       
+        verify(postRequestedFor(urlEqualTo("/api/v5/projects/1/routes-stats"))
```

Comment:
As part of this stub can we use something like a [withRequestBody](https://wiremock.org/docs/request-matching/) call here here so we can verify that we are providing the correct payload?